src/main/java/com/inorg/services/product/service/ProductServiceImpl.java

```diff
@@ -131,20 +188,132 @@ public List<Product> createProducts() {
 
     @Override
     public ProductPagedQueryResponse getProductsByCategory(String categoryId) {
-        return null;
-        // apiRoot.products().get()
+       // return null;
+        return  apiRoot.products().get()
+                .addWhere("masterData(current(categories(id = \""+categoryId+"\")))")
                 //TODO withWhere
-                //.executeBlocking().getBody();
+                .executeBlocking().getBody();
     }
 
     @Override
     public ProductProjectionPagedSearchResponse searchProducts(String searchText) {
-        return null;
-       //  apiRoot.productProjections().search()
-                //TODO withText, withFacet
-                //.executeBlocking().getBody();
+        //return null;
+        //TODO withText withFacet
+         return apiRoot.productProjections().search()
+                 .get()
+                 .withText("EN-US", searchText)
+                 .withFacet("variants.attributes.Color:\""+searchText+"\"")
+                 .withFacet("variants.attributes.Size:\""+searchText+"\"")
+                .executeBlocking().getBody();
+    }
+
+    @Override
+    public ProductProjectionPagedQueryResponse getProductBySKU(String sku) {
+        return apiRoot.productProjections()
+                .get()
+                .addWhere("masterVariant(sku=\""+sku+"\") or variants(sku=\""+sku+"\")")
+                .executeBlocking()
+                .getBody();
+    }
+
+    @Override
+    public Product updateProductPriceWithSKU(String sku, String price) {
+
+        Product productToBeUpdate = null;
+        ProductProjectionPagedQueryResponse product = null;
+        product =  getProductBySKU(sku);
+        if(product.getResults().isEmpty()){
+            System.out.println("No Product found ");
+        } else {
+            System.out.println("Product found");
+            AtomicReference<String> priceId = new AtomicReference<>();
```

Comment:
One Cleaner approach using java streams

`List<Product> returnProducts = new ArrayList<>();
        productProjectionPagedQueryResponse.getResults().forEach(productProjection -> {

            List<String> priceIds = new ArrayList<>();
            productProjection.getMasterVariant().getPrices().stream()
                    .filter(price -> price.getValue().getCurrencyCode().equals("USD"))
                    .forEach(price -> {
                        priceIds.add(price.getId());
                    });
            productProjection.getVariants().forEach(productVariant -> {
                productVariant.getPrices().stream()
                        .filter(price -> price.getValue().getCurrencyCode().equals("USD"))
                        .forEach(price -> {
                            priceIds.add(price.getId());
                        });
            });

            List<ProductUpdateAction> updateActions = new ArrayList<>();
            priceIds.forEach(priceId -> {
                updateActions.add(ProductChangePriceActionBuilder.of()
                        .priceId(priceId)
                        .price(PriceDraftBuilder.of()
                                .value(MoneyBuilder.of()
                                        .currencyCode("USD")
                                        .centAmount(Long.parseLong(price))
                                        .build())
                                .build())
                        .build()
                );
            });

            ProductUpdate productUpdate = ;

            returnProducts.add(
                    apiRoot.products()
                        .withId(productProjection.getId())
                        .post(ProductUpdate.builder()
                                .version(productProjection.getVersion())
                                .actions(updateActions)
                                .build())
                        .executeBlocking()
                        .getBody());
        });

        return returnProducts;`